course_catalog/etl/youtube_test.py

```diff
@@ -0,0 +1,189 @@
+                    "publishedAt": "2019-08-30T04:00:05.000Z",
+                    "thumbnails": {},
+                    "title": "Pitch and Roll and Prince Charles",
+                },
+            }
+        ],
+        "kind": "youtube#searchListResponse",
+        "pageInfo": {"resultsPerPage": 50, "totalResults": 51},
+        "regionCode": "US",
+    }
+
+    if token != "":
+        data["nextPageToken"] = token
+
+    return data
+
+
+@pytest.fixture
+def mock_video_data():
+    """Mock data for a single video"""
+    data = {
+        "contentDetails": {
+            "caption": "true",
+            "definition": "hd",
+            "dimension": "2d",
+            "duration": "PT3M14S",
+            "licensedContent": False,
+            "projection": "rectangular",
+        },
+        "etag": '"j6xRRd8dTPVVptg711_CSPADRfg/Bh4Z5YJX4kTKyCsVio3xqVS9nx0"',
+        "id": "QU0fLnucE6A",
+        "kind": "youtube#video",
+        "snippet": {
+            "categoryId": "27",
+            "channelId": "UCTBMWu8yshnAmpzR3MoJFtw",
+            "channelTitle": "MITx Videos",
+            "defaultAudioLanguage": "en",
+            "description": "Professor Krishna Rajagopal explains Faraday Cages and why it's not such a bad thing if you're in a lightning storm as long as you're inside a car. Learn more and enroll now at https://www.edx.org/course/electricity-and-magnetism-electrostatics?utm_campaign=mitx&utm_medium=partner-marketing&utm_source=social&utm_content=youtube-8.01.1x-faraday",
+            "liveBroadcastContent": "none",
+            "localized": {
+                "description": "Professor Krishna Rajagopal explains Faraday Cages and why it's not such a bad thing if you're in a lightning storm as long as you're inside a car. Learn more and enroll now at https://www.edx.org/course/electricity-and-magnetism-electrostatics?utm_campaign=mitx&utm_medium=partner-marketing&utm_source=social&utm_content=youtube-8.01.1x-faraday",
+                "title": "Faraday Cage",
+            },
+            "publishedAt": "2018-10-09T18:18:09.000Z",
+            "thumbnails": {},
+        },
+    }
+
+    return data
+
+
+@pytest.fixture
+def mock_video_batch_data(mock_video_data):
+    """Mock video request data for a batch"""
+    data = {
+        "kind": "youtube#videoListResponse",
+        "etag": '"j6xRRd8dTPVVptg711_CSPADRfg/j0RF7LBD1f8ViAQN-D94fJ40WZ8"',
+        "pageInfo": {"totalResults": 1, "resultsPerPage": 1},
+        "items": [mock_video_data],
+    }
+
+    return data
+
+
+@pytest.fixture
+def mock_channel_data():
+    """Mock video channel yml"""
+    data = """
+---
+channels:
+  - offered_by: MIT
+    channel_id: channel1
+  - offered_by: OCW
+    channel_id: channel2
+"""
+
+    return data
+
+
+@pytest.fixture
+def mocked_channel_response(mocked_responses, mock_channel_data, settings):
+    """Mock the channel catalog response"""
+    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube.yaml"
```

Comment:
I set settings.VIDEOS_CONFIG_FILE_LOCATION both here and line 130 which seems wrong but when I take out this line the mock does not have access to the value set in line 130. I tried a few things, and none of them worked. 

I think I can DRY up the tests a bit more in general and can use some help doing that.

Reply:
I think `pytests`'s autouse fixtures will help you here. It will get applied to every test in the file. You can define one like this:

```python
@pytest.fixture(autouse=True)
def video_settings(settings):
    settings.VIDEOS_CONFIG_FILE_LOCATION = "https://youtube_empty.yaml"
    settings.YT_DEVELOPER_KEY = "key"
    return settings
```

Then you can use `video_settings` instead of `settings` in your function/fixture parameters to ensure this fixture is run first.

Reply:
Thanks! That works